### ISSUE TITLE
Build with JDK 11+ as a workaround for JDK-8190452

### ISSUE THREAD

**Issue by Stephan202** [classgraph/classgraph#577]
As of d841f638a23faeb81dd43270887b1cceef29e2a9, part of version 4.8.121, ClassGraph is compiled with `-parameters`. Prior to Java 11 this feature may produce strictly-speaking invalid class files due to the issue described in [JDK-8190452](https://bugs.openjdk.java.net/browse/JDK-8190452). ClassGraph users who compile with `-Xlint:classfile` or `-Xlint:all` will now see this compiler warning:
```
[WARN] MethodParameters attribute introduced in version 52.0 class files is ignored in version 51.0 class files
```

In combination with `-Werror` this causes a compilation failure. This currently prevents us from upgrading beyond version 4.8.120. For a similar case see google/dagger#945.

I verified that the issue can be resolved by compiling the code with JDK 11+:
- I can locally reproduce the issue by building a SNAPSHOT from `latest` using JDK 8.
- When I rebuild the code using JDK 11 our downstream build no longer emits the aforementioned warning.

The relevant compiler output differences between JDK 8 and JDK 11 can be seen as follows:
```sh
# With JDK 1.8.0_302
$ find . -name "*.class" -print0 | xargs -0 javap -p -v | grep MethodParameters | wc -l
1718
# With JDK 11.0.12
$ find . -name "*.class" -print0 | xargs -0 javap -p -v | grep MethodParameters | wc -l
245
```

**Comment by lukehutch** [comment]
Thanks for the report, I didn't realize this would cause a problem! I had a user reporting that tests were failing in JDK 17 on Windows (#562), and it appears that Oracle decided to switch on the `-parameters` switch by default on that JDK build (which seems odd, but maybe it is the new normal). This caused tests to fail, because the tests were written to use `toString` comparisons for `MethodInfo` etc., for simplicity.

Since `-parameters` is only needed when building unit tests, I just moved the compiler parameter to the execution of the test build. This should solve the problem, but please confirm whether this fixes the issue for you in the latest git version.

**Comment by Stephan202** [comment]
Thanks for the quick response @lukehutch!

I just built `latest` (rev dfc1b0596b725c1663eacf253344f7a8469f01c9) using JDK 8 and can confirm that the issue is resolved :+1:

**Comment by lukehutch** [comment]
Awesome, likewise thanks to you for the quick response! Fixed and released in 4.8.126.

**Comment by lukehutch** [comment]
@Stephan202 I also added `-Xlint:all -Werror` to the default compiler arguments, so that this sort of issue is caught in the future. Thanks for the report!
